lib/success_response.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'json'
+
+class SuccessResponse
+  attr_reader :error_message
+
+  def initialize(response)
+    @response = response
+    @error_message = nil
+  end
+
+  def ok?
+    body
+    true
```

Comment:
You are parsing JSON, which can error. 

Now you have to check **two** things, `ok?` and `error_message`.
Or completely remove the `ok?` method, which now is useless.

Or you can refactor how I told you (just add a bang before the error in the `ok?` method: `!error`).

Please make sure you understand my other comment.

Reply:
> You are parsing JSON, which can error.

You've said previously 

> Any 2XX code is considered a success, not only 200.

>thinking but why? all 2XX are success even 201

> and always return true from ok? in the SuccessResponse. After all, it's the success response, it's always going to be OK.
and in our chat:

> all you need is a hardcoded true in ok? and a hash in body



> Now you have to check two things, ok? and error_message.
You are assuming that I would use the error_message to check (I don't plain to)

> Or completely remove the ok? method, which now is useless.
you said in our chat:

> type checks like instance_of are a code smell remember Sandy Metz, you want to use a duck type
>  otherwise what you are doing is "if I know which type you are, I will provide the behaviour"
> that's why you have the ok? method, so that you don't have to check the type



> just add a bang before the error in the ok? method: !error
on of the main reasons why I wouldn't want to do that is that I think it would make the overall code in that class difficult to read.

Reply:
maybe it will help you to understand my intentions with how I can handle the returns now:

`def manage_sql_update
  return client.body unless client.ok?  # this returns a string

  return sql.update(client.body) unless client.body.empty? # this will return a 'success' string or failure string

  client.error_message  # this returns a string
end`